File: Reboot/SlidingWindow/MaximumSumSubarraySizeK.py
def maximumSubSubarraySizeK(arr: list, k: int) -> int:
    if len(arr) < k:
        return 0
    
    i: int = 0
    maxSum: int = 0
    currentSum: int = 0
    for j in range(len(arr)):
        currentSum += arr[j]
        if j - i + 1 < k:
            continue
        else:
            maxSum = max(currentSum, maxSum)
            currentSum -= arr[i]
            i += 1
    return maxSum

arr: list[int] = [1, 4, 2, 10, 2, 3, 1, 0, 20]
k: int = 3
print("Maximum Subarrau sum of size k: ", maximumSubSubarraySizeK(arr=arr, k=k))

arr: list[int] = [100, 200, 300, 400]
k: int = 4
print("Maximum Subarrau sum of size k: ", maximumSubSubarraySizeK(arr=arr, k=k))

File: Reboot/SlidingWindow/MaximumSumSubarraySizeK.cpp
#include<stdio.h>
#include<iostream>
using namespace std;

int maximumSubSubarraySizeK(int arr[], int k, int n) {
    int i = 0;
    int currentSum = 0;
    int maxSum = 0;
    for (int j = 0; j < n; j++) {
        currentSum += arr[j];
        if ((j - i + 1) < k) {
            continue;
        } else {
            maxSum = max(currentSum, maxSum);
            currentSum -= arr[i];
            i += 1;
        }
    }
    return maxSum;
}

int main() {
    int arr[] = {1, 4, 2, 10, 2, 3, 1, 0, 20};
    int k = 3;
    int n = sizeof(arr)/sizeof(arr[0]);
    cout << "Maximum Subarray sum of size k: " << maximumSubSubarraySizeK(arr, k, n) << endl;    
    return 0;
}